-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39459/32581

ERROR: Build errors found during clang-tidy run.
```
/cvmfs/cms-ib.cern.ch/nweek-02754/el8_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-10-14-1100/src/FWCore/Framework/interface/maker/WorkerMaker.h:71:8: error: call to non-static member function without an object argument [clang-diagnostic-error]
    T::fillDescriptions(iDesc);
       ^
/cvmfs/cms-ib.cern.ch/nweek-02754/el8_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-10-14-1100/src/FWCore/Framework/interface/maker/WorkerMaker.h:55:14: note: in instantiation of member function 'edm::WorkerMaker<LHEGenericMassFilter>::fillDescriptions' requested here
--
/cvmfs/cms-ib.cern.ch/nweek-02754/el8_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-10-14-1100/src/FWCore/ParameterSet/interface/ParameterSetDescriptionFiller.h:40:10: error: call to non-static member function without an object argument [clang-diagnostic-error]
      T::fillDescriptions(descriptions);
         ^
/cvmfs/cms-ib.cern.ch/nweek-02754/el8_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-10-14-1100/src/FWCore/ParameterSet/interface/ParameterSetDescriptionFiller.h:35:5: note: in instantiation of member function 'edm::ParameterSetDescriptionFiller<LHEGenericMassFilter>::fill' requested here
--
GeneratorInterface/GenFilters/plugins/LHEGenericMassFilter.cc:91:28: error: out-of-line definition of 'fillDescriptions' does not match any declaration in 'LHEGenericMassFilter' [clang-diagnostic-error]
void LHEGenericMassFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
                           ^
GeneratorInterface/GenFilters/plugins/LHEGenericMassFilter.cc:94:8: error: no matching member function for call to 'add' [clang-diagnostic-error]
  desc.add<std::vector<int>>("ParticleID", 1);
       ^
/cvmfs/cms-ib.cern.ch/nweek-02754/el8_amd64_gcc10/cms/cmssw-patch/CMSSW_12_6_X_2022-10-14-1100/src/FWCore/ParameterSet/interface/ParameterSetDescription.h:95:31: note: candidate function template not viable: no known conversion from 'int' to 'const std::vector<int>' for 2nd argument
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/GeneratorInterface/GenFilters/plugins/LHEGenericMassFilter.cc
+++ b/GeneratorInterface/GenFilters/plugins/LHEGenericMassFilter.cc
@@ -24,6 +24,7 @@ class LHEGenericMassFilter : public edm::global::EDFilter<> {
 public:
   explicit LHEGenericMassFilter(const edm::ParameterSet&);
   ~LHEGenericMassFilter() override = default;
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
 
 private:
   bool filter(edm::StreamID, edm::Event&, edm::EventSetup const&) const override;
@@ -91,7 +92,7 @@ bool LHEGenericMassFilter::filter(edm::StreamID iID, edm::Event& iEvent, edm::Ev
 void LHEGenericMassFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
   edm::ParameterSetDescription desc;
   desc.add<int>("NumRequired", 1);
-  desc.add<std::vector<int>>("ParticleID", 1);
+  desc.add<vector<int>>("ParticleID", std::vector<int>{1});
   desc.add<double>("MinMass", 0.0);
   desc.add<double>("MaxMass", 1.0);
   desc.add<bool>("RequiredOutgoingStatus", true);

--- a/GeneratorInterface/GenFilters/plugins/LHEGenericMassFilter.cc
+++ b/GeneratorInterface/GenFilters/plugins/LHEGenericMassFilter.cc
@@ -42,7 +42,7 @@ private:
 LHEGenericMassFilter::LHEGenericMassFilter(const edm::ParameterSet& iConfig)
     : src_(consumes<LHEEventProduct>(iConfig.getParameter<edm::InputTag>("src"))),
       numRequired_(iConfig.getParameter<int>("NumRequired")),
-      particleID_(iConfig.getParameter<std::vector<int> >("ParticleID")),
+      particleID_(iConfig.getParameter<std::vector<int>>("ParticleID")),
       minMass_(iConfig.getParameter<double>("MinMass")),
       maxMass_(iConfig.getParameter<double>("MaxMass")),
       requiredOutgoingStatus_(iConfig.getParameter<bool>("RequiredOutgoingStatus")) {}
@@ -61,7 +61,7 @@ bool LHEGenericMassFilter::filter(edm::StreamID iID, edm::Event& iEvent, edm::Ev
 
   for (int i = 0; i < EvtHandle->hepeup().NUP; ++i) {
     // if requiredOutgoingStatus_ keep only outgoing particles, otherwise keep them all
-    if (requiredOutgoingStatus_ && EvtHandle->hepeup().ISTUP[i] != 1) {  
+    if (requiredOutgoingStatus_ && EvtHandle->hepeup().ISTUP[i] != 1) {
       continue;
     }
     for (unsigned int j = 0; j < particleID_.size(); ++j) {

